base_layer/core/src/proof_of_work/lwma_diff.rs

```diff
@@ -87,7 +87,15 @@ impl LinearWeightedMovingAverage {
             .skip(1)
             .fold(0u128, |difficulty, (_, d)| difficulty + u128::from(d.as_u64()));
 
-        let ave_difficulty = difficulty / n;
+        let ave_difficulty = if let Some(ave) = difficulty.checked_div(n) {
```

Comment:
`n` has already been checked above and can never be 0. `n = self.target_difficulties.len() - 1` where  `self.target_difficulties.len() must be at least 2.

Reply:
Right, I didn't realize that. So these changes are not needed actually :)